### DISCUSSION TITLE
Delete Disks / volumes

### DISCUSSION THREAD

**Discussion by m1cha3lf** [ClickHouse/ClickHouse#58728] [Q&A]
Hi,

i'm having a clickhouse instance running with 3 disks.

```
SELECT * FROM system.disks;

d2,/ssd/disk2/clickhouse/,500435656704,527296159744,500435656704,0,local,0,0,0,0,0,""
d3,/mnt/ssd2/disk3/clickhouse/,99703734272,105089261568,99703734272,0,local,0,0,0,0,0,""
default,/var/lib/clickhouse/,303549431808,1040448036864,65545498617,0,local,0,0,0,0,0,""

````

I'd like to remove disk `d2` permanently. All tables who had a storage policy pointing to d2 are deleted already. 

I tried to remove the disk from storage.xml, but it doesn't get deleted.  Whatever i try to remove the storage it ends with clickhouse not starting due to some depdenencies on the disk `d2`.

What is the correct way to remove a disk permanently?

Thank in advance

Michael

**Comment by mlex** [comment]
Hi!
Can you share your `storage.xml`? And the error message when clickhouse refuses to start after you remove `d2`?
